include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -735,11 +742,14 @@ class binary_writer
     @param[in] use_type    whether to use '$' prefixes (optimized format)
     @param[in] add_prefix  whether prefixes need to be used for this value
     @param[in] use_bjdata  whether write in BJData format, default is false
+    @param[in] bjdata_version  which BJData version to use, default is draft2
     */
     void write_ubjson(const BasicJsonType& j, const bool use_count,
                       const bool use_type, const bool add_prefix = true,
-                      const bool use_bjdata = false)
+                      const bool use_bjdata = false, const bjdata_version_t bjdata_version = bjdata_version_t::draft2)
     {
+        const bool bjdata_draft3 = bjdata_version == bjdata_version_t::draft3;
```

Comment:
If you make this `const bool bjdata_draft3 = use_bjdata && bjdata_version == bjdata_version_t::draft3;` then you can eliminate the `use_bjdata &&` on each of the checks below, and in some cases the extra parens around that.

Reply:
Fixed in https://github.com/nlohmann/json/pull/4588